auth/auth.go

```diff
@@ -156,7 +156,8 @@ func NewAuthenticator(ctx context.Context, c *Config) (*Authenticator, error) {
 		switch c.AuthSource {
 		case AuthSourceOpenShift:
 			// Use the k8s CA for OAuth metadata discovery.
-			client, err := newHTTPClient(c.DiscoveryCA, false)
```

Comment:
This was masking the `err` var from above, so it was always evaluating to `nil` when we check below for failures